The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8879743373/job/24378213841) [(plain)](https://github.com/rust-lang/rust/commit/f67a0da9059417baaecc4575c8cf2ef03aa14ada/checks/24378213841/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:0ad4b8fadaa221de15dcec353f45205ec38ea70b)
Download action repository 'actions/upload-artifact@v4' (SHA:65462800fd760344b1a7b4382951275a0abb4808)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.419   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.431 Collecting boolean-py==4.0
#13 0.435   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.445 Collecting chardet==5.1.0
---
#13 3.439 Building wheels for collected packages: reuse
#13 3.440   Building wheel for reuse (pyproject.toml): started
#13 3.761   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.762   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 3.762   Stored in directory: /tmp/pip-ephem-wheel-cache-l_ds90rb/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 3.764 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 3.787   Attempting uninstall: setuptools
#13 3.788     Found existing installation: setuptools 59.6.0
#13 3.789     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 3.789     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 3.789     Can't uninstall 'setuptools'. No files were found to uninstall.
#13 4.460 Successfully installed binaryornot-0.4.4 boolean-py-4.0 chardet-5.1.0 jinja2-3.1.2 license-expression-30.0.0 markupsafe-2.1.1 python-debian-0.1.49 reuse-1.1.0 setuptools-66.0.0
#13 4.460 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.970 Collecting virtualenv
#13 5.007   Downloading virtualenv-20.26.0-py3-none-any.whl (3.9 MB)
#13 5.062      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.9/3.9 MB 74.3 MB/s eta 0:00:00
#13 5.113 Collecting platformdirs<5,>=3.9.1
#13 5.117   Downloading platformdirs-4.2.1-py3-none-any.whl (17 kB)
#13 5.150 Collecting filelock<4,>=3.12.2
#13 5.153   Downloading filelock-3.13.4-py3-none-any.whl (11 kB)
#13 5.170 Collecting distlib<1,>=0.3.7
#13 5.173   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.179      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 113.1 MB/s eta 0:00:00
#13 5.268 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.433 Successfully installed distlib-0.3.8 filelock-3.13.4 platformdirs-4.2.1 virtualenv-20.26.0
#13 DONE 5.5s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      188352 kB
DirectMap2M:     7151616 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished `dev` profile [unoptimized] target(s) in 0.03s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/90846015cc99aa90290c1f4ee95c571fff6901ef/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-90846015cc99aa90290c1f4ee95c571fff6901ef-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 24.90s
##[endgroup]
fmt check
##[error]Diff in /checkout/compiler/rustc_lint/src/context.rs at line 159:
                 // Don't display deprecated lint groups.
                 depr.is_none()
             })
-            .map(|(k, LintGroup { lint_ids, is_externally_loaded, .. })| (*k, lint_ids.clone(), *is_externally_loaded))
+            .map(|(k, LintGroup { lint_ids, is_externally_loaded, .. })| {
+                (*k, lint_ids.clone(), *is_externally_loaded)
     }
 
     pub fn register_early_pass(
     pub fn register_early_pass(
Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/compiler/rustc_driver_impl/src/lib.rs" "/checkout/compiler/rustc_driver_impl/src/print.rs" "/checkout/compiler/rustc_driver_impl/src/signal_handler.rs" "/checkout/compiler/rustc_driver_impl/src/pretty.rs" "/checkout/compiler/rustc_driver_impl/src/session_diagnostics.rs" "/checkout/compiler/rustc_driver_impl/src/args.rs" "/checkout/compiler/rustc_driver/src/lib.rs" "/checkout/compiler/rustc_lint/src/invalid_from_utf8.rs" "/checkout/compiler/rustc_lint/src/foreign_modules.rs" "/checkout/compiler/rustc_lint/src/unused.rs" "/checkout/compiler/rustc_lint/src/errors.rs" "/checkout/compiler/rustc_lint/src/context/diagnostics/check_cfg.rs" "/checkout/compiler/rustc_lint/src/context/diagnostics.rs" "/checkout/compiler/rustc_lint/src/opaque_hidden_inferred_bound.rs" "/checkout/compiler/rustc_lint/src/enum_intrinsics_non_enums.rs" "/checkout/compiler/rustc_lint/src/non_ascii_idents.rs" "/checkout/compiler/rustc_lint/src/passes.rs" "/checkout/compiler/rustc_lint/src/array_into_iter.rs" "/checkout/compiler/rustc_lint/src/types.rs" "/checkout/compiler/rustc_lint/src/late.rs" "/checkout/compiler/rustc_lint/src/lints.rs" "/checkout/compiler/rustc_lint/src/async_fn_in_trait.rs" "/checkout/compiler/rustc_lint/src/levels.rs" "/checkout/compiler/rustc_lint/src/lib.rs" "/checkout/compiler/rustc_lint/src/methods.rs" "/checkout/compiler/rustc_lint/src/early.rs" "/checkout/compiler/rustc_lint/src/builtin.rs" "/checkout/compiler/rustc_lint/src/internal.rs" "/checkout/compiler/rustc_lint/src/deref_into_dyn_supertrait.rs" "/checkout/compiler/rustc_lint/src/for_loops_over_fallibles.rs" "/checkout/compiler/rustc_lint/src/redundant_semicolon.rs" "/checkout/compiler/rustc_lint/src/non_fmt_panic.rs" "/checkout/compiler/rustc_lint/src/ptr_nulls.rs" "/checkout/compiler/rustc_lint/src/expect.rs" "/checkout/compiler/rustc_lint/src/noop_method_call.rs" "/checkout/compiler/rustc_lint/src/nonstandard_style.rs" "/checkout/compiler/rustc_lint/src/drop_forget_useless.rs" "/checkout/compiler/rustc_lint/src/pass_by_value.rs" "/checkout/compiler/rustc_lint/src/reference_casting.rs" "/checkout/compiler/rustc_lint/src/traits.rs" "/checkout/compiler/rustc_lint/src/unit_bindings.rs" "/checkout/compiler/rustc_lint/src/let_underscore.rs" "/checkout/compiler/rustc_lint/src/nonstandard_style/tests.rs" "/checkout/compiler/rustc_lint/src/hidden_unicode_codepoints.rs" "/checkout/compiler/rustc_lint/src/context.rs" "/checkout/compiler/rustc_lint/src/non_local_def.rs" "/checkout/compiler/rustc_lint/src/map_unit_fn.rs" "/checkout/compiler/rustc_lint/src/tests.rs" "/checkout/compiler/rustc_lint/src/multiple_supertrait_upcastable.rs" "/checkout/compiler/rustc_ty_utils/src/consts.rs" "/checkout/compiler/rustc_ty_utils/src/errors.rs" "/checkout/compiler/rustc_ty_utils/src/needs_drop.rs" "/checkout/compiler/rustc_ty_utils/src/lib.rs" "/checkout/compiler/rustc_ty_utils/src/structural_match.rs" "/checkout/compiler/rustc_ty_utils/src/implied_bounds.rs" "/checkout/compiler/rustc_ty_utils/src/layout.rs" "/checkout/compiler/rustc_ty_utils/src/ty.rs" "/checkout/compiler/rustc_ty_utils/src/opaque_types.rs" "/checkout/compiler/rustc_ty_utils/src/assoc.rs" "/checkout/compiler/rustc_ty_utils/src/abi.rs" "/checkout/compiler/rustc_ty_utils/src/representability.rs" "/checkout/compiler/rustc_ty_utils/src/layout_sanity_check.rs" "/checkout/compiler/rustc_ty_utils/src/sig_types.rs" "/checkout/compiler/rustc_target/src/lib.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Mon Apr 29 13:55:13 UTC 2024
  network time: Mon, 29 Apr 2024 13:55:13 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/compiler/rustc_driver_impl/src/lib.rs
+++ b/compiler/rustc_driver_impl/src/lib.rs
@@ -971,7 +971,7 @@ Available lint options:
 
     let lint_store = unerased_lint_store(sess);
     let (loaded, builtin): (Vec<_>, _) =
-        lint_store.get_lints().iter().cloned().partition(|&lint| lint.is_loaded);
+        lint_store.get_lints().iter().cloned().partition(|&lint| lint.is_externally_loaded);
     let loaded = sort_lints(sess, loaded);
     let builtin = sort_lints(sess, builtin);
 

--- a/compiler/rustc_lint/src/context.rs
+++ b/compiler/rustc_lint/src/context.rs
@@ -110,7 +110,7 @@ struct LintAlias {
 
 struct LintGroup {
     lint_ids: Vec<LintId>,
-    is_loaded: bool,
+    is_externally_loaded: bool,
     depr: Option<LintAlias>,
 }
 
@@ -159,7 +159,9 @@ impl LintStore {
                 // Don't display deprecated lint groups.
                 depr.is_none()
             })
-            .map(|(k, LintGroup { lint_ids, is_loaded, .. })| (*k, lint_ids.clone(), *is_loaded))
+            .map(|(k, LintGroup { lint_ids, is_externally_loaded, .. })| {
+                (*k, lint_ids.clone(), *is_externally_loaded)
+            })
     }
 
     pub fn register_early_pass(
@@ -218,7 +220,7 @@ impl LintStore {
                         .entry(edition.lint_name())
                         .or_insert(LintGroup {
                             lint_ids: vec![],
-                            is_loaded: lint.is_loaded,
+                            is_externally_loaded: lint.is_externally_loaded,
                             depr: None,
                         })
                         .lint_ids
@@ -231,7 +233,7 @@ impl LintStore {
                         .entry("future_incompatible")
                         .or_insert(LintGroup {
                             lint_ids: vec![],
-                            is_loaded: lint.is_loaded,
+                            is_externally_loaded: lint.is_externally_loaded,
                             depr: None,
                         })
                         .lint_ids
@@ -246,7 +248,7 @@ impl LintStore {
             alias,
             LintGroup {
                 lint_ids: vec![],
-                is_loaded: false,
+                is_externally_loaded: false,
                 depr: Some(LintAlias { name: lint_name, silent: true }),
             },
         );
@@ -254,21 +256,21 @@ impl LintStore {
 
     pub fn register_group(
         &mut self,
-        is_loaded: bool,
+        is_externally_loaded: bool,
         name: &'static str,
         deprecated_name: Option<&'static str>,
         to: Vec<LintId>,
     ) {
         let new = self
             .lint_groups
-            .insert(name, LintGroup { lint_ids: to, is_loaded, depr: None })
+            .insert(name, LintGroup { lint_ids: to, is_externally_loaded, depr: None })
             .is_none();
         if let Some(deprecated) = deprecated_name {
             self.lint_groups.insert(
                 deprecated,
                 LintGroup {
                     lint_ids: vec![],
-                    is_loaded,
+                    is_externally_loaded,
                     depr: Some(LintAlias { name, silent: false }),
                 },
             );

--- a/compiler/rustc_lint_defs/src/lib.rs
+++ b/compiler/rustc_lint_defs/src/lib.rs
@@ -323,7 +323,8 @@ pub struct Lint {
 
     pub future_incompatible: Option<FutureIncompatibleInfo>,
 
-    pub is_loaded: bool,
+    /// `true` if this lint is being loaded by another tool (e.g. Clippy).
+    pub is_externally_loaded: bool,
 
     /// `Some` if this lint is feature gated, otherwise `None`.
     pub feature_gate: Option<Symbol>,
@@ -468,7 +469,7 @@ impl Lint {
             default_level: Level::Forbid,
             desc: "",
             edition_lint_opts: None,
-            is_loaded: false,
+            is_externally_loaded: false,
             report_in_external_macro: false,
             future_incompatible: None,
             feature_gate: None,
@@ -817,7 +818,7 @@ macro_rules! declare_lint {
             name: stringify!($NAME),
             default_level: $crate::$Level,
             desc: $desc,
-            is_loaded: false,
+            is_externally_loaded: false,
             $($v: true,)*
             $(feature_gate: Some($gate),)?
             $(future_incompatible: Some($crate::FutureIncompatibleInfo {
@@ -859,7 +860,7 @@ macro_rules! declare_tool_lint {
             edition_lint_opts: None,
             report_in_external_macro: $external,
             future_incompatible: None,
-            is_loaded: true,
+            is_externally_loaded: true,
             $(feature_gate: Some($gate),)?
             crate_level_only: false,
             ..$crate::Lint::default_fields_for_macro()

--- a/tests/ui/statics/nested_struct.rs
+++ b/tests/ui/statics/nested_struct.rs
@@ -9,7 +9,7 @@ pub struct Lint {
     pub name: &'static str,
     pub desc: &'static str,
     pub report_in_external_macro: bool,
-    pub is_loaded: bool,
+    pub is_externally_loaded: bool,
     pub crate_level_only: bool,
 }
 
@@ -17,7 +17,7 @@ static FOO: &Lint = &Lint {
     name: &"foo",
     desc: "desc",
     report_in_external_macro: false,
-    is_loaded: true,
+    is_externally_loaded: true,
     crate_level_only: false,
 };
 

